beacon_node/client/src/builder.rs

```diff
@@ -195,7 +195,22 @@ where
             None
         };
 
-        let builder = BeaconChainBuilder::new(eth_spec_instance)
+        let kzg = if let Some(trusted_setup) = config.clone().trusted_setup {
+            let kzg_err_msg = |e| format!("Failed to load trusted setup: {:?}", e);
+
+            if spec.is_peer_das_scheduled() {
+                Arc::new(
+                    Kzg::new_from_trusted_setup_das_enabled(trusted_setup).map_err(kzg_err_msg)?,
+                )
+            } else {
+                Arc::new(Kzg::new_from_trusted_setup(trusted_setup).map_err(kzg_err_msg)?)
+            }
+        } else {
+            // TODO(kzg) this causes us failed tests i.e. http_server_genesis_state
+            panic!("Failed to load KZG");
```

Comment:
this panic causes some of our tests to fail i.e. `http_server_genesis_state`. I'm assuming a trusted setup isnt available pre-deneb. So for tests that run a pre-deneb network, is there some default KZG value we want to pass in here?

Reply:
I think we should try loading the KZG setup regardless of whether the network is pre-Deneb (if possible).

Reply:
Wouldn't that slow things significantly? Post PeerDAS loading KZG is slow given the precomputation optimization

Reply:
Oh. Hmm. Maybe we need to find a way to only load it once in tests, as otherwise we'll be loading it 1000s of times, even if we just load it for peer DAS tests?

Reply:
A new feature in 0.5.0 allows a user to specify whether they want precomputation or not: https://github.com/crate-crypto/rust-eth-kzg/pull/161

Its under breaking changes: `Allow downstream users to avoid the precomputations that are needed for the FixedBaseMSM algorithm in FK20` since it adds a new parameter that was not there before.

I can push another release today/tomorrow if that would help -- Noting that if you want the same computing and verifying speeds without precomputation, you can increase the number of threads you initialize the library with to something like 4 instead of 1

Reply:
thanks Kev! That new feature sounds like exactly what we need here. I'll try it out and report back if I'm having issues

Reply:
0.5 includes a few API changes, so it might be better if I made the PR so as to not waste your time, if this blocking for this PR :)